packages/smart-vaults/src/utils/OperatorManager.sol

```diff
@@ -0,0 +1,199 @@
+    error TargetCannotBeAccount();
+
+    /* -------------------------------------------------------------------------- */
+    /*                                  MODIFIERS                                 */
+    /* -------------------------------------------------------------------------- */
+
+    modifier onlyOperator() {
+        if (!_getOperatorManagerStorage().isOperator[msg.sender]) revert OnlyOperator();
+        _;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                          EXTERNAL/PUBLIC FUNCTIONS                         */
+    /* -------------------------------------------------------------------------- */
+
+    /**
+     * @notice Adds operator to the allowlist.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     */
+    function addOperator(address operator_) public {
+        _authorize();
+
+        _addOperator(operator_);
+    }
+
+    /**
+     * @notice Adds `operator` to the allowlist and makes a call to the `setupContract` passing `data_`.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     * @param setupContract_ address of contract to call to setup operator.
+     * @param data_ data passed to the setupContract.
+     */
+    function addAndSetupOperator(address operator_, address setupContract_, bytes calldata data_) public {
+        _authorize();
+
+        _addOperator(operator_);
+
+        _call(setupContract_, 0, data_);
+    }
+
+    /**
+     * @notice Removes operator from the allowlist.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     */
+    function removeOperator(address operator_) public {
+        _authorize();
+
+        _removeOperator(operator_);
+    }
+
+    /**
+     * @notice Removes `operator` from the allowlist and makes a call to the `teardownContract_` passing `data_`.
+     *
+     * @dev Can only be called by this account.
+     *
+     * @param operator_ address of operator.
+     * @param teardownContract_ address of contract to call to teardown operator.
+     * @param data_ data passed to the setupContract.
+     */
+    function removeAndTeardownOperator(address operator_, address teardownContract_, bytes calldata data_) public {
+        _authorize();
+
+        _removeOperator(operator_);
+
+        _call(teardownContract_, 0, data_);
+    }
+
+    /**
+     * @notice Executes a single call from the account.
+     *
+     * @dev Can only be called by the operator.
+     * @dev Reverts if Call.target is account (address(this)).
+     * @dev Emits an event to capture the call executed.
+     *
+     * @param call_ Call to execute from the account.
+     */
+    function executeFromOperator(Call calldata call_) external onlyOperator {
+        if (call_.target == address(this)) revert TargetCannotBeAccount();
```

Comment:
Disabled this thinking we wouldn't want an operator to be able to perform owner operations that are on the account itself and the only function they should be able to call is executeFromOperator but I guess it doesn't make sense since they can perform any action from the account